src/embind/embind.js

```diff
@@ -694,8 +694,10 @@ var LibraryEmbind = {
             // assumes 4-byte alignment
             var length = getLength();
             var ptr = _malloc(4 + length + 1);
+#if CAN_ADDRESS_2GB
```

Comment:
Instead of having a `CAN_ADDRESS_2GB` option, could we reuse `WASM_MEM_MAX` option? (`#if WASM_MEM_MAX >= 2GB`)? (if not, would it better be `CAN_ADDRESS_4GB`?)

Reply:
We can get rid of the setting in settings.js, but then we'd need to compute "can we refer to address 2GB+" in both JS and Python separately. It's not a hard computation, though - check if growth or not, and if yes, what the limit is and if there is no limit. I can do it either way, but an internal setting seems like less code and more uniform between the two world (JS and Python).

I think `CAN_ADDRESS_2GB` is more accurate, as this problem exists for people doing 2GB + 64KB, 3GB, etc., and not just 4GB. They key issue is at address 2GB and above, pointers start to use the high bit.

(`WASM_MEM_MAX` was renamed `MAXIMUM_MEMORY`, btw - but old one still works)